[jit] class_table_ to deps_table_

suo opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #23799 [wip] serializing function calls
* #23846 [jit] clean up import_source
* **#23845 [jit] class_table_ to deps_table_**
* #23697 [jit] make FunctionType a NamedType
* #23696 [jit] make NamedType an interface
* #23691 [jit] simplify NamedType interface
* #23689 [jit] search class type for methods

These are not just classes anymore, rename

Differential Revision: [D16684391](https://our.internmc.facebook.com/intern/diff/D16684391)

suo committed:
[jit] class_table_ to deps_table_

These are not just classes anymore, rename

ZolotukhinM commented on torch/csrc/jit/passes/python_print.cpp:
```diff
@@ -272,18 +272,17 @@ struct PythonPrintPass {
   // dependencies.
-  std::vector<c10::NamedTypePtr>& class_table_;
-  std::vector<c10::NamedTypePtr> direct_class_deps_;
+  std::vector<c10::NamedTypePtr>& deps_table_;
```
Remove the comment above?

ZolotukhinM commented on torch/csrc/jit/export.cpp:
```diff
@@ -621,7 +619,7 @@ void ScriptModuleSerializer::writeLibs(torch::ModelDef* model_def) {
 
   // Convert all the classes that this model depends on
-  for (const auto& class_type : class_table_) {
+  for (const auto& class_type : deps_table_) {
```
Could `deps_table_` contain something `convertClass` doesn't expect?

ZolotukhinM reviewed (approved):
Could you please elaborate in the commit message what a dependency can be other than a class?

Otherwise looks good!

suo committed:
Update on "[jit] class_table_ to deps_table_"


These are not just classes anymore, rename

suo committed:
Update on "[jit] class_table_ to deps_table_"


These are not just classes anymore, rename

suo committed:
Update on "[jit] class_table_ to deps_table_"


These are not just classes anymore, rename

Differential Revision: [D16684391](https://our.internmc.facebook.com/intern/diff/D16684391)

suo committed:
Update on "[jit] class_table_ to deps_table_"


These are not just classes anymore, rename

Differential Revision: [D16684391](https://our.internmc.facebook.com/intern/diff/D16684391)

suo committed:
Update on "[jit] class_table_ to deps_table_"


These are not just classes anymore, rename

Differential Revision: [D16684391](https://our.internmc.facebook.com/intern/diff/D16684391)
